dsgrid/registry/dataset_registry_manager.py

```diff
@@ -127,37 +143,123 @@ def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup
                     f"load_data_lookup records do not match dimension records for {name}"
                 )
 
-    def _check_dataset_time_consistency(self, config: DatasetConfig, load_data):
-        time_dim = config.get_dimension(DimensionType.TIME)
-        tz = time_dim.get_tzinfo()
-        time_ranges = time_dim.get_time_ranges()
-        assert len(time_ranges) == 1, len(time_ranges)
-        time_range = time_ranges[0]
-        # TODO: need to support validation of multiple time ranges: DSGRID-173
-
-        expected_timestamps = time_dim.list_time_range(time_range)
-        actual_timestamps = [
-            x.timestamp.astimezone().astimezone(tz)
-            for x in load_data.select("timestamp").distinct().sort("timestamp").collect()
-        ]
-        if expected_timestamps != actual_timestamps:
-            mismatch = sorted(
-                set(expected_timestamps).symmetric_difference(set(actual_timestamps))
+    def _check_one_table_data_consistency(self, config: DatasetConfig, load_data_one_table):
+        """
+        Check all data dimensions, including Metric records.
+        For Time, only check that time_col_name = timestamp
+        """
+        dimension_types = []
+        metric_cols = []
+
+        for col in load_data_one_table.columns:
+            if col.lower() == "time":
+                raise ValueError(
+                    f"col_name={col} for the time dimension is not supported, used col_name='timestamp' instead."
+                )
+            else:
+                try:
+                    dimension_types.append(DimensionType(col))
+                except ValueError:
+                    if col != "timestamp":
+                        metric_cols.append(col)
```

Comment:
Do we know what columns are required ahead of time? If so, it would be better to handle them explicitly. It seems like this implicit handling could be problematic. Should an exception be raised if a column is neither a valid dimension nor a valid metric?